src/Avalonia.Base/Utilities/ArrayBuilder.cs

```diff
@@ -182,13 +215,12 @@ private void EnsureCapacity(int min)
         /// <returns>The <see cref="ArraySlice{T}"/>.</returns>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public ArraySlice<T> AsSlice(int start, int length) => new ArraySlice<T>(_data!, start, length);
-        
-        public void Dispose()
-        {
-            if (_data != null)
-            {
-                ArrayPool<T>.Shared.Return(_data);
```

Comment:
We should think about some typical data size and if we surpass that size we should reset the data array. Otherwise, the array will grow to the longest processed text until the application is terminated. This was the main reason I did remove the ThreadLocal instance.

Reply:
That's right, I'll add that right away. (the callers should specify the limit since they are the ones that should know what they want). There are also a few reusable `List<T>` to reset if they grow too large.

Reply:
I just pushed a new commit that adds the limit.

The threshold is for now an arbitrary 1MB. It's a 512KB text _without any like break_, which is quite unlikely to happen, but I know to never underestimate the users :)

If needed, it can be made configurable later.